ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Types.hs

```diff
@@ -195,6 +199,13 @@ data ChainDbEnv m blk = CDB
     --
     -- Note that the \"immutable\" block will /never/ be /more/ than @k@
     -- blocks back, as opposed to the anchor point of 'cdbChain'.
+  , cdbTentativeState  :: !(StrictTVar m (TentativeState blk))
+  , cdbTentativeHeader :: !(StrictTVar m (Maybe (Header blk)))
```

Comment:
@amesgen We should use a `MaybeHeader` type here that is strict.

Also, the Haddock is stale: we do read from it as of your promptness fix

Reply:
Esgen noted that I neglected to pull before rebasing. That explains the stale comment. But the `TVar Maybe` remains.

I have re-rebased (it was entirely automatic this time).